app/services/record_declarations/base.rb

```diff
@@ -39,7 +39,10 @@ def call
       validate_provider!
       validate_milestone!
 
+      raise ActiveRecord::RecordNotUnique, "Declaration with given participant ID already exists" if record_exist?
+
       declaration = find_or_create_record!
+
```

Comment:
Should we do something like:
```
declaration = find_or_create_record! # make it more lenient - find by things we use in record_exists?
validate_declaration_uniqueness!(declaration)
...

def validate_declaration_uniqueness!(declaration)
  # check date
  # check provider
  # check anything else that we want to check
  # raise if the checks fail
end
```

Reply:
The reason I think it should be done before the declaration creation as otherwise we end up with duplicate declarations after ```find_or_create_record``` but in the ticket requirement we don't want to do that. What do you think?